src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -8,18 +8,20 @@
 import zipfile
 from contextlib import contextmanager
 
+from pants.backend.jvm.targets.jar_library import JarLibrary
 from pants.build_graph.aliased_target import AliasTarget
 from pants.build_graph.target import Target
 from pants.util.contextutil import open_zip
 
 
 def _resolve_strict_dependencies(target):
```

Comment:
I'd like to see a comment that discusses the reasoning behind this. In particular, an alternative would have been to require `exports` on `jar_library` targets in order for them to expose their deps. In the long run, that might better align with `exports` being supported for more target types.

It's possible that having `strict_deps` not apply to `JarLibrary` is something that we should put behind an option, and not do by default.